src/aws/osml/model_runner/database/feature_table.py

```diff
@@ -121,72 +136,71 @@ def add_features(self, features: List[Feature], metrics: MetricsLogger = None):
-                        metrics.put_metric(MetricLabels.FEATURE_ERROR, 1, str(Unit.COUNT.value))
-                    logger.error("There was a problem adding features: {}".format(err))
-                    raise AddFeaturesException("Failed to add features for tile!") from err
+                # Write batch to the table and check that it succeeded for all items
+                self.batch_write_items(items)
+
+            except Exception as err:
+                if isinstance(metrics, MetricsLogger):
+                    metrics.put_metric(MetricLabels.FEATURE_UPDATE_EXCEPTION, 1, str(Unit.COUNT.value))
+                    metrics.put_metric(MetricLabels.FEATURE_ERROR, 1, str(Unit.COUNT.value))
+                raise AddFeaturesException("Failed to add features for tile!") from err
 
     @metric_scope
     def get_features(self, image_id: str, metrics: MetricsLogger = None) -> List[Feature]:
         """
-        Query the database for all items with given image_id, the convert them into feature items, then
-        go through all of the items and group the features per tile
+        Parallelized version to query the database for all items with a given image_id,
+        then convert them into feature items, and group the features per tile.
 
-        :param image_id: str = unique image_id for the job
-        :param dedupe: Optional[bool] = remove any duplicate items
+        :param image_id: The image_id to aggregate features from DDB for.
         :param metrics: MetricsLogger = the metrics logger to use to report metrics.
-
-        :return: List[Feature] = the list of features
+        :return: List of features aggregates from the DDB table.
         """
+
+        def process_query(index: int):
+            items: List[FeatureItem] = []
+            rows = self.query_items(FeatureItem(image_id + "-" + str(index)))
+            for row in rows:
+                items.append(from_dict(FeatureItem, row))
+            return items
+
         if isinstance(metrics, MetricsLogger):
             metrics.set_dimensions()
+
+        features: List[Feature] = []
+
         with Timer(
             task_str="Aggregate image features",
             metric_name=MetricLabels.FEATURE_AGG_LATENCY,
             logger=logger,
             metrics_logger=metrics,
         ):
-            # Query the database for all items with the given image_id (hash_key)
-            rows = self.query_items(FeatureItem(image_id))
+            with ThreadPoolExecutor(max_workers=5) as executor:
+                # Create a range of tasks to query the database for the salted image hash
+                futures = [executor.submit(process_query, i) for i in range(1, self.hash_salt)]
+
+                # For each of the salted index processes the returned items
+                for future in as_completed(futures):
+                    feature_items = future.result()
+                    grouped_items = self.group_items_by_tile_id(feature_items)
```

Comment:
As you pointed out in slack this call to group_items_by_tile_id is now unnecessary and can be skipped. Just iterate over all the feature_items and append them to batch_features.